lib/widgets-customize.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Bootstrapping the Gutenberg widgets editor in the customizer.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Gutenberg's Customize Register.
+ *
+ * Adds a section to the Customizer for editing widgets with Gutenberg.
+ *
+ * @param \WP_Customize_Manager $manager An instance of the class that controls most of the Theme Customization API for WordPress 3.4 and newer.
+ */
+function gutenberg_widgets_customize_register( $manager ) {
+	global $wp_registered_sidebars;
+
+	if ( ! gutenberg_use_widgets_block_editor() ) {
+		return;
+	}
+
+	require_once __DIR__ . '/class-wp-sidebar-block-editor-control.php';
+
+	foreach ( $manager->sections() as $section ) {
+		if ( $section instanceof WP_Customize_Sidebar_Section ) {
+			$section->description = '';
+		}
+	}
+	foreach ( $manager->controls() as $control ) {
+		if (
+			$control instanceof WP_Widget_Area_Customize_Control ||
+			$control instanceof WP_Widget_Form_Customize_Control
+		) {
+			$manager->remove_control( $control->id );
+		}
+	}
+
+	foreach ( $wp_registered_sidebars as $sidebar_id => $sidebar ) {
+		$manager->add_setting(
```

Comment:
Is there not already a widget area setting? Curious why we're adding a new one.

Reply:
I thought it would be a good practice to create a new setting?

Reply:
Cool 👍, just wondering if you ran into something specific. We will eventually want a new setting so that we can change how widgets are serialised.